### DISCUSSION TITLE
umi3内置antd，但是没有联想

### DISCUSSION THREAD

**Discussion by Luxiaoshi** [umijs/umi#8383] [Q&A]
# umi3内置antd，但是没有联想
umi@3.5.26
之前用umi@2的时候，是有联想的。类似在tsx里输入In，会自动联想出antd的Input。但是目前发现好像不行
包括umi的history也不行。
是需要做额外的配置吗
<img width="493" alt="image" src="https://user-images.githubusercontent.com/8713879/177245401-0bb3b42b-abc0-417c-bbd8-90a676e1604f.png">

**Comment by fz6m** [comment]
没有 antd 就装一下 antd 呗

**Reply by Luxiaoshi** [reply]
自带的antd，确实是有的，但是就是没有联想
<img width="458" alt="image" src="https://user-images.githubusercontent.com/8713879/177294048-6a3cad50-2453-43f2-b6eb-deed5e5de01e.png">

**Reply by fz6m** [reply]
装了才会有，因为 antd 不在你的 `package.json` 里，IDE 没法联想。

**Reply by Luxiaoshi** [reply]
nodemodules里是有antd的，umi应该是通过插件的形式内置了antd，所以应该不必重复安装吧？

**Reply by fz6m** [reply]
1. 如果你不安装 antd 用的是自带的 antd ，如果安装了就会用你自己的
2. IDE 不会根据你 node_modules 有没有给你代码提示，要明确安装
3. 综上，你这种情况需要自己安装 antd
